### PR TITLE
PodFitsResources Predicate Metadata as prefilter

### LINKED ISSUE #86248: Move pod fits resources predicate metadata into pre-filter plugin [Migration Phase 2]
/assign @ahg-g 
/sig scheduling
/priority important-soon

Part of #85822

### PR DESCRIPTION
**Pull request by ahg-g**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Wraps NodeResourcesFit metadata in a prefilter.

**Which issue(s) this PR fixes**:
Fixes #86248

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @Huang-Wei

### CODE REVIEW

**Comment by ahg-g** [comment]
Note that this PR converted ignoredExtendedResources to an argument rather than a metadata member.

File: pkg/scheduler/factory.go [hunk 1]
```diff
@@ -192,13 +193,15 @@ func (c *Configurator) CreateFromConfig(policy schedulerapi.Policy) (*Scheduler,
 			}
 			for _, r := range policy.Extenders[ii].ManagedResources {
 				if r.IgnoredByScheduler {
-					ignoredExtendedResources.Insert(string(r.Name))
+					ignoredExtendedResources = append(ignoredExtendedResources, string(r.Name))
```
Comment (Huang-Wei) [inline pkg/scheduler/factory.go]:
```suggestion
					ignoredExtendedResources = append(ignoredExtendedResources, r.Name)
```

File: pkg/scheduler/framework/plugins/noderesources/fit.go [hunk 2]
```diff
@@ -20,38 +20,107 @@ import (
...
+	args := &FitArgs{}
+	if err := framework.DecodeInto(plArgs, args); err != nil {
+		return nil, err
+	}
+
+	fit := &Fit{}
+	fit.ignoredResources = sets.NewString()
+	for _, r := range args.IgnoredResources {
+		fit.ignoredResources.Insert(r)
+	}
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/noderesources/fit.go]:
```go
fit.ignoredResources = sets.NewString(args.IgnoredResources...)
```

**Review by Huang-Wei — COMMENTED** [review]
Just some nits.

File: pkg/scheduler/framework/plugins/noderesources/fit.go (hunk 2 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/noderesources/fit.go]: why? insert seems more readable.

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/noderesources/fit.go]: done.

File: pkg/scheduler/factory.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/factory.go]: done.

**Review by ahg-g — COMMENTED** [review]
rebased

**Pull request merged**